src/app/authentication/service/otpService.go

```diff
@@ -0,0 +1,77 @@
+package service
+
+import (
+	"Stock_broker_application/repo"
+	"crypto/rand"
+	"encoding/base64"
+	"errors"
+	"time"
+	// "time"
+)
+
+type OTPService struct {
+    UserRepository repo.UserRepository
+}
+
+func NewOTPService(userRepository repo.UserRepository) *OTPService {
+    return &OTPService{
+        UserRepository: userRepository,
+    }
+}
+
+// GenerateAndSaveOTP generates a random OTP, saves it along with its creation time in the database
+// for the given email address
+func (s *OTPService) GenerateAndSaveOTP(email string) error {
+    // Generate random OTP
+    otp, err := generateRandomOTP()
+    if err != nil {
+        return err
+    }
+
+  
+	
+    if err := s.UserRepository.SaveOTP(email, otp); err != nil {
+        return err
+    }
+
+    return nil
+}
+
+// generateRandomOTP generates a random 6-digit OTP
+func generateRandomOTP() (string, error) {
+    // Generate 3 bytes of random data
+    bytes := make([]byte, 3)
+    if _, err := rand.Read(bytes); err != nil {
+        return "", err
+    }
+
+    // Encode the bytes to base64
+    otp := base64.StdEncoding.EncodeToString(bytes)
+
+    // Strip any special characters from the OTP
+    otp = otp[:4]
```

Comment:
implement better way to generate 4/6 digit random OTP, which only gives numeric and not characters.